src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java

```diff
@@ -509,6 +513,32 @@ public void beforeCommand(CommandEnvironment env) throws AbruptExitException {
             invocationId,
             remoteOptions.remoteInstanceName));
 
+    ActionResultDownloader actionResultDownloader = null;
+    Preconditions.checkState(remoteOutputService == null, "remoteOutputService must be null");
+    if (remoteOptions.remoteOutputService != null) {
+      try {
+        ReferenceCountedChannel channel =
+            RemoteCacheClientFactory.createGrpcChannelPool(
+                channelFactory,
+                poolSize,
+                remoteOptions.remoteOutputService,
+                remoteOptions.remoteProxy,
+                authAndTlsOptions,
```

Comment:
I believe this will mean the same auth and tls options will be used when contacting remote output service and remote execution service parts (like CAS?). Not sure whether this may break in some setups, like if someone will use tls to connect to CAS, then ROS will need to be deployed with proper server certificate as well so that client will trust ROS.

Reply:
At least for the bb_clientd use case this is exactly what's needed. bb_clientd's FUSE file system also needs to have credentials for the Remote Execution service, so that it can load files that are accessed through the file system. By letting Bazel send Remote Execution credentials to bb_clientd, it is able to reuse the credentials that are configured on the Bazel side.

(Furthermore, bb_clientd can act like a full REv2 proxy, so that a system-wide cache of blobs may be offered.)